src/main/java/sparksoniq/jsoniq/compiler/parser/Jsoniq.g4

```diff
@@ -167,9 +167,15 @@ jSONItemTest
 keyWordBoolean
          : 'boolean';
 
-atomicType
+singleType
+         : item=simpleType (question +='?')?;
+
+simpleType
          : 'atomic' | 'string' | 'integer'
-         | 'decimal' | 'double' | keyWordBoolean | NullLiteral;
+         | 'decimal' | 'double' | keyWordBoolean;
+
+atomicType
+         : simpleType | NullLiteral;
```

Comment:
Actually null behaves like any other atomic type: it is a single type mapping the literal "null" to null, and null("null") returns null, "null" castable as null returns true. This was done so as not to introduce any special treatment for it. There is no need to introduce any special treatment by making the grammar more complicated and splitting atomicType in this respect.

Reply:
I will reply here to this and subsequent comments on adding _NullItem_ to the list of item types we can cast to. 

From the second review iteration, I added it to the AtomicTypes (as it should have been since the beginning), I removed the SimpleType definition in the grammar, and added the necessary checks on whether items can be cast to NullItem and viceversa. Indeed now, we can do something like _x castable as null_. Only _null castable as null_, _"null" castable as null_ and the same queries with the _ZeroOrOne_ arity on the target item type, returns true. All other castable_as queries with null as target item type will return false. Also, _null castable as string_ will be true.